terraform/eval_validate.go

```diff
@@ -101,6 +121,64 @@ func (n *EvalValidateProvisioner) Eval(ctx EvalContext) (interface{}, error) {
 	}
 }
 
+func (n *EvalValidateProvisioner) validateConnConfig(connConfig *ResourceConfig) (warns []string, errs []error) {
```

Comment:
Ugh... it turns out that this is more problematic than I initially thought because the `communicator` package already depends on the `terraform` package and so we can't call into it from `terraform` without creating a dependency cycle.

We've got away with this so far because it's the provisioners that call into `communicator` and the plugin indirection prevents Go from seeing it as a circular dependency.

I think there is a path here where the `communicator` interface can be adjusted so that it receives directly the connection info as a `map[string]interface{}` rather than getting a `terraform.InstanceState` and pulling it out of there itself, but that's a bit beyond the scope of what I wanted to do for this bug.

Do you think it's reasonable to move ahead with what I had here and address this later, or should I just shelve this for the moment and come back to it when there's more time for such refactoring?